### PR TITLE
Improve controller and unit tests

### PR DESCRIPTION
**Pull request by alingrig87**
- improve GreetingsController to contain examples for GET/POST/PUT/DELETE
- add unit tests

image

### CODE REVIEW

File: src/test/java/ro/unibuc/hello/dto/GreetingTest.java [hunk 1]
```diff
@@ -5,15 +5,15 @@
 
 class GreetingTest {
 
-    Greeting myGreeting = new Greeting(1, "John");
+    Greeting myGreeting = new Greeting("1", "John");
```
Comment (vgrigoriu) [inline src/test/java/ro/unibuc/hello/dto/GreetingTest.java:8]: Not sure if this test pulls its weight, we can probably remove it in a separate PR.

File: src/main/java/ro/unibuc/hello/service/GreetingsService.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return new Greeting(Long.toString(counter.incrementAndGet()), String.format(informationTemplate, entity.getTitle(), entity.getDescription()));
+    }
+
+    public Greeting saveGreeting(Greeting greeting) {
+        InformationEntity entity = new InformationEntity();
+        entity.setId(greeting.getId());
+        entity.setTitle(greeting.getContent());
+        informationRepository.save(entity);
+        return new Greeting(entity.getId(), entity.getTitle());
+    }
```
Comment (vgrigoriu) [inline src/main/java/ro/unibuc/hello/service/GreetingsService.java:41]: I find this a bit confusing, we're actually saying we save/update/delete/etc. `Greeting`s, but we actually persist `InformationEntity`es. I mean sure, let's isolate the types we store in the DB from the types we use in the public API, but I think in this case the connection between `Greeting` and `InformationEntity` is not clear.

File: src/test/java/ro/unibuc/hello/service/HelloWorldServiceTestIT.java
Comment (vgrigoriu) [inline src/test/java/ro/unibuc/hello/service/HelloWorldServiceTestIT.java:1]: Don't we also want a `GreetingsServiceIntegrationTest`, to show off the difference between unit and integration tests?

**Review by vgrigoriu — APPROVED** [review]
Thanks @alingrig87 for taking the time to add CRUD operations and unit tests for them. I left some comments but I think it's fine to merge this and address them later.

File: src/test/java/ro/unibuc/hello/service/HelloWorldServiceTestIT.java
Comment (alingrig87) [reply src/test/java/ro/unibuc/hello/service/HelloWorldServiceTestIT.java:1]: Yes, we want to add integration tests, but in another PR.

File: src/main/java/ro/unibuc/hello/service/GreetingsService.java (hunk 2 above)
Comment (alingrig87) [reply src/main/java/ro/unibuc/hello/service/GreetingsService.java:41]: Me too. :)

**Comment by alexOporanu1** [comment]
Looks good to me! Thanks for working on this!

**Review by alexOporanu1 — APPROVED** [review]

**Pull request merged**